source/xlsxreader.d

```diff
@@ -160,67 +170,40 @@ struct Sheet {
 		mixin(format(t2, T, T[0].toUpper ~ T[1 .. $]));
 	}
 
-	RowUntyped iterateRowUntyped(size_t row, size_t startColumn, size_t endColumn) @safe {
+	RowUntyped iterateRowUntyped(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(T) iterateRow(T)(size_t row, size_t startColumn, size_t endColumn) @trusted /* TODO: remove @trusted when `&this` is stored in a @safe manner in `Row` */ {
+	Row!(T) iterateRow(T)(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(long) iterateRowLong(size_t row, size_t startColumn, size_t endColumn) @safe {
+	Row!(long) iterateRowLong(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(double) iterateRowDouble(size_t row, size_t startColumn, size_t endColumn) @safe {
+	Row!(double) iterateRowDouble(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(string) iterateRowString(size_t row, size_t startColumn, size_t endColumn) @safe {
+	Row!(string) iterateRowString(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(DateTime) iterateRowDateTime(size_t row, size_t startColumn, size_t endColumn) @safe {
+	Row!(DateTime) iterateRowDateTime(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(Date) iterateRowDate(size_t row, size_t startColumn, size_t endColumn) @safe {
+	Row!(Date) iterateRowDate(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 
-	Row!(TimeOfDay) iterateRowTimeOfDay(size_t row, size_t startColumn, size_t endColumn) @safe {
+	Row!(TimeOfDay) iterateRowTimeOfDay(size_t row, size_t startColumn, size_t endColumn) @safe return {
 		return typeof(return)(&this, row, startColumn, endColumn);
 	}
 }
 
-struct Iterator(T) {
-	T[] data;
-
-	this(T[] data) {
-		this.data = data;
-	}
-
-	@property bool empty() const pure nothrow @nogc {
-		return this.data.empty;
-	}
-
-	void popFront() {
-		this.data.popFront();
-	}
-
-	@property T front() {
-		return this.data.front;
-	}
-
-	inout(typeof(this)) save() inout pure nothrow @nogc {
-		return this;
-	}
-
-	// Request random access.
-	inout(T)[] array() inout @safe pure nothrow @nogc {
-		return data;
-	}
-}
+alias Iterator(T) = T[];
```

Comment:
I think this alias can be deleted now.

It's all arrays as far as I can see.

Reply:
I left it in for somewhat less breakage, in case a library has been using `Iterator`.